AM62D OOB: Audio-DSP-Offload Linux example

v-singh1 opened the pull request.

cshilwant commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+The incoming audio stream is assumed to be sampled at 48KHz. Below figure shows how this demo works:
+
+
+        .. Image:: /images/AM62D_DSP_offload_Demo_Setup.png
```
Image and figure directives needs to have a width and height value pre-defined. Eg - https://github.com/TexasInstruments/processor-sdk-doc/blob/master/CONTRIBUTING.md#indentation-and-whitespace

Also, better to switch to `figure` directive for all the instances where we are adding images to be consistent

v-singh1 replied:
Switched to figure

v-singh1 replied:
Fixed

cshilwant reviewed (changes requested):
Apart from the above comments, commemts from vale and commit check workflow needs to be addressed

vale warnings at - https://github.com/TexasInstruments/processor-sdk-doc/pull/355/files#diff-c3ca6facf690bd9373acdba648b491b7255559ecbe93abb8a115e47f861924c8

commit check error at - https://github.com/TexasInstruments/processor-sdk-doc/actions/runs/16195790963/job/45721822147?pr=355

`feat(Demo Application): Add Audio-DSP-Offload Linux example` as commit message title should be fine. Also, a short commit message can be added

v-singh1 commented on source/images/AM62D_com_ports.png:
Agree, removed it.

v-singh1 replied:
Agree, removed

v-singh1 commented on source/images/AM62D_linux_login.jpg:
Not required, removed it

v-singh1 replied:
Not required, removed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+#. Make sure the EVM boot mode switches are set properly for SD card boot as described earlier
+
+#. Connect the USB-C cable from the power adapter to one of the two USB-C ports on the EVM.i
```
fixed

v-singh1 replied:
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+   - For more information refer: `README <https://github.com/TexasInstruments/rpmsg-dma/blob/scarthgap/example/audio_offload/host%20utility/README.md>`_.
+
+   Note: Input/output audio spectrum plotting is only supported in ip mode. UART mode supports only metrics and command interface, not audio data streams.
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+      root@am62dxx-evm:~# rpmsg_audio_offload_example
+
+#. On host machine launch the audiomon.py utility either in IP mode or UART mode
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+Building Audio DSP offload wic image from Yocto
+-----------------------------------------------
+
+#. To build the Audio DSP offload wic image, please refer :ref:`Processor SDK - Building the SDK with Yocto <building-the-sdk-with-yocto>`
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+   - Real-time visualization of:
+
+        - Frame-level average amplitude (dBFS)
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+        .. code-block:: console
+
+           host# git clone https://github.com/TexasInstruments/rpmsg-dma.git
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,270 @@
+
+      root@am62dxx-evm:~# ifconfig
+
+- Note: EVM ip address is required for host utility to connect to demo application
```
Fixed

cshilwant commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,265 @@
+Building the C7 Firmware from sources
+--------------------------------------
+
+- Please refer to the `MCU+ SDK Documentation  <https://software-dl.ti.com/mcu-plus-sdk/esd/AM62DX/11_00_00_16/exports/docs/api_guide_am62dx/GETTING_STARTED_BUILD.html>`__
```
@jeevantelukula / @v-singh1  A note to update this link to 11.1 MCU+ SDK version once it's live

v-singh1 replied:
Yes, update is planned once 11.1 MCU SDK is live

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,265 @@
+   :height: 600
+   :width: 1200
+
+- For more information on demo application & it's configuration, refer:  `DSP Offload Example <https://github.com/TexasInstruments/rpmsg-dma/blob/scarthgap/example/audio_offload/host%20utility/README.md>`__.
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,265 @@
+
+   - For more information refer: `README <https://github.com/TexasInstruments/rpmsg-dma/blob/scarthgap/example/audio_offload/host%20utility/README.md>`_.
+
+.. note:: Note: Input/output audio spectrum plotting is only supported in ip mode. UART mode supports only metrics and command interface, not audio data streams.
```
Fixed

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,265 @@
+
+      root@am62dxx-evm:~# ifconfig
+
+.. Note:: EVM ip address is required for host utility to connect to demo application
```
Fixed

v-singh1 commented:
@StaticRocket @praneethbajjuri can you please review.

v-singh1 commented:
> Can you use SVG for graphics where possible?
> 
> As for images, the "Host GUI Utility" image has multiple of the same step number, but none of this is explained.
> 
> It also uses the word "Date" instead of "data".

Addressed review comments.

v-singh1 commented:
> As initially requested please address the vale comments on spelling, active voice, etc.
> 
> You may ignore the note about common abbreviations.
> 
> Also, you'll need to remove the excess `:` in your commit. Proper formatting would be `feat(demos): AM62D Audio-DSP-Offload Linux example`

Addressed review comments, acronyms and  below 2  warnings are left as they are required.

[vale] reported by reviewdog 🐶
[RedHat.Headings] Use sentence-style capitalization in 'AM62D DSP offload from Linux - User Guide'.

[vale] reported by reviewdog 🐶
[RedHat.SimpleWords] Use simple language. Consider using 'tell' rather than 'Notify'.

StaticRocket commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+
+    - The example application :file:`rpmsg_audio_offload_example`
+
+  - Transfer the generated files to evm sdcard:
```
```suggestion
  - Transfer the generated files to evm SD card:
```

v-singh1 replied:
Done

StaticRocket commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+
+.. note::
+   
+   Input/output audio spectrum plotting is only supported in ip mode. UART mode supports only metrics and command interface, not audio data streams.
```
```suggestion
   Input/output audio spectrum plotting is only supported in IP mode. UART mode supports only metrics and command interface, not audio data streams.
```

v-singh1 replied:
Done

StaticRocket commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+
+#. The EVM should boot and the booting progress should display in the serial port console. At the end of booting, the Arago login prompt will appear. Just enter "root" to log in.
+
+#. Get the EVM ip address
```
```suggestion
#. Get the EVM IP address
```

v-singh1 replied:
Done

StaticRocket commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+
+.. note::
+
+   EVM ip address required for host utility to connect to demo application
```
```suggestion
   EVM IP address required for host utility to connect to demo application
```

v-singh1 replied:
Done

StaticRocket reviewed (changes requested):
Some spelling issues that were truncated due to API rate limits from other warnings

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+- Step 5: Signal chain execution on DSP
+   - Below figure shows the C7x signal chain executed on C7x:
+
+   .. figure:: /images/AM62D_dsp_signal_chain_11.01.png
```
Done

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+    - Windows: Windows 11 64-bit
+
+    - Linux: Ubuntu 22.04 64-bit or higher
+      
```
Done

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+       
+     - BOOTMODE [ 0 : 7 ] (SW2) = 1100 0010
+
+.. figure:: /images/AM62D_evm_sdcard_boot_mode.png
```
Done

v-singh1 commented on source/linux/Demo_User_Guides/AM62D_Dsp_Offload_User_Guide.rst:
```diff
@@ -0,0 +1,298 @@
+Steps to validate audio DSP offload demo
+----------------------------------------
+
+#. Flash an SD card with the :file:`tisdk-default-image-rt-am62dxx-evm.rootfs.wix.xz` wic image and follow the instructions provided at :ref:`Create SD Card <processor-sdk-linux-create-sd-card>` guide.
```
Done
